### PR TITLE
Fix the test failing due to timestamp localisation

### PR DESCRIPTION
**Pull request by Gu1nness**
Fixes issue #4853

### CODE REVIEW

File: src/borg/testsuite/archiver.py [hunk 1]
```diff
@@ -2552,15 +2553,20 @@ def test_recreate_recompress(self):
...
         archive = self.repository_location + '::test0'
         self.cmd('create', archive, 'input')
         self.cmd('recreate', '--timestamp', "1970-01-02T00:00:00", '--comment',
                  'test', archive)
         info = self.cmd('info', archive).splitlines()
-        assert any([re.search(r'Time \(start\).+ 1970-01-02', item) for item in info])
-        assert any([re.search(r'Time \(end\).+ 1970-01-02', item) for item in info])
+        dtime = datetime(1970, 1, 2) + local_timezone.utcoffset(None)
+        assert any([re.search(r'Time \(start\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
+        assert any([re.search(r'Time \(end\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: see my comment in the ticket.

File: src/borg/testsuite/archiver.py [hunk 2]
```diff
@@ -2552,15 +2553,20 @@ def test_recreate_recompress(self):
         assert sha256_before == sha256_after
 
     def test_recreate_timestamp(self):
+        if sys.version_info >= (3, 6):
+            local_timezone = datetime.now(timezone(timedelta(0))).astimezone().tzinfo
+        else:
+            local_timezone = datetime.now(timezone.utc).astimezone().tzinfo
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: is there a way to do this without checking the python version?

File: src/borg/testsuite/archiver.py (hunk 1 above)
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: can you move the strftime call to outside the loops, only computing it once after line 2567?

File: src/borg/testsuite/archiver.py [hunk 3]
```diff
@@ -2553,10 +2553,7 @@ def test_recreate_recompress(self):
         assert sha256_before == sha256_after
 
     def test_recreate_timestamp(self):
-        if sys.version_info >= (3, 6):
-            local_timezone = datetime.now(timezone(timedelta(0))).astimezone().tzinfo
-        else:
-            local_timezone = datetime.now(timezone.utc).astimezone().tzinfo
+        local_timezone = datetime.now(timezone(timedelta(0))).astimezone().tzinfo
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: so, this also works for py35?

File: src/borg/testsuite/archiver.py [hunk 4]
```diff
@@ -2565,8 +2562,9 @@ def test_recreate_timestamp(self):
                  'test', archive)
         info = self.cmd('info', archive).splitlines()
         dtime = datetime(1970, 1, 2) + local_timezone.utcoffset(None)
-        assert any([re.search(r'Time \(start\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
-        assert any([re.search(r'Time \(end\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
+        s_time = dtime.strftime("%Y-%m-%d")
+        assert any([re.search(r'Time \(start\).+ %s' % s_time , item) for item in info])
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: pep8 checker is unhappy:

src/borg/testsuite/archiver.py:2566:62: E203 whitespace before ','

File: src/borg/testsuite/archiver.py (hunk 3 above)
Comment (Gu1nness) [reply src/borg/testsuite/archiver.py]: According to my tests and Travis, yes :)

File: src/borg/testsuite/archiver.py (hunk 4 above)
Comment (ThomasWaldmann) [reply src/borg/testsuite/archiver.py]: fixed now.

**Comment by ThomasWaldmann** [comment]
i played a bit with datetime, tried your code and thought about the problem again.

from https://borgbackup.readthedocs.io/en/stable/usage/general.html#date-and-time
```
Unless otherwise noted, we display local date and time.
Internally, we store and process date and time as UTC.
```
OK, that does not directly say something about `--timestamp`, but as the cli is also part of the user interface (as output is), this might mean that maybe not the test needs fixing, but our processing of that option.

**But, the docs also say**:
```
--timestamp TIMESTAMP
manually specify the archive creation date/time (UTC,  yyyy-mm-ddThh:mm:ss format).
Alternatively, give a reference  file/directory.
```

So, using UTC there is a documented feature, although it is different from output using localtime.

So, the test fix in this PR is correct according to the docs. \o/

**Comment by ThomasWaldmann** [comment]
thanks! can you check if 1.1-maint has the same issue?

**Comment by Gu1nness** [comment]
Yes, see #4873 for the commit fixing this on 1.1-maint

**Pull request merged**